packages/smithy-core/src/smithy_core/aio/client.py

```diff
@@ -0,0 +1,567 @@
+                response=response_context.transport_response,
+                error_registry=call.operation.error_registry,
+                context=response_context.properties,
+            )
+
+            _LOGGER.debug("Deserialization complete. Output: %s", output)
+
+            output_context = OutputContext(
+                request=response_context.request,
+                response=output,
+                transport_request=response_context.transport_request,
+                transport_response=response_context.transport_response,
+                properties=response_context.properties,
+            )
+
+            interceptor.read_after_deserialization(output_context)
+        except Exception as e:
+            output_context = OutputContext(
+                request=request_context.request,
+                response=e,
+                transport_request=request_context.transport_request,
+                transport_response=None,
+                properties=request_context.properties,
+            )
+
+        return self._finalize_attempt(call, output_context)
+
+    def _resolve_auth[I: SerializeableShape, O: DeserializeableShape](
+        self, call: ClientCall[Any, Any], params: AuthParams[I, O]
+    ) -> tuple[AuthOption, AuthScheme[TRequest, Any, Any, Any]] | None:
+        auth_options: Sequence[AuthOption] = (
+            call.auth_scheme_resolver.resolve_auth_scheme(auth_parameters=params)
+        )
+
+        for option in auth_options:
+            if (
+                scheme := call.supported_auth_schemes.get(option.scheme_id)
+            ) is not None:
+                return option, scheme
+
+        return None
+
+    def _finalize_attempt[I: SerializeableShape, O: DeserializeableShape](
+        self,
+        call: ClientCall[I, O],
+        output_context: OutputContext[I, O, TRequest, TResponse | None],
+    ) -> OutputContext[I, O, TRequest, TResponse | None]:
+        interceptor = call.interceptor
+        try:
+            output_context = replace(
+                output_context,
+                response=interceptor.modify_before_attempt_completion(output_context),
+            )
+        except Exception as e:
+            output_context = replace(output_context, response=e)
+
+        try:
+            interceptor.read_after_attempt(output_context)
+        except Exception as e:
+            output_context = replace(output_context, response=e)
+
+        return output_context
+
+    def _finalize_execution[I: SerializeableShape, O: DeserializeableShape](
+        self,
+        call: ClientCall[I, O],
+        output_context: OutputContext[I, O, TRequest | None, TResponse | None],
+    ) -> OutputContext[I, O, TRequest | None, TResponse | None]:
+        interceptor = call.interceptor
+        try:
+            output_context = replace(
+                output_context,
+                response=interceptor.modify_before_completion(output_context),
+            )
+
+            # TODO trace probe
```

Comment:
I don't think we're noting this anywhere else. Was there something specific that prompted this TODO?

Reply:
Are we just calling out doing [this block](https://github.com/smithy-lang/smithy-python/blob/4bcdb752773ddc67ffe5def457c1bf1c53160561/codegen/core/src/main/java/software/amazon/smithy/python/codegen/ClientGenerator.java#L751-L757)?